### PR TITLE
WIP / Handle complex provenance merging better.

### PR DESCRIPTION
**Pull request by sellout**

### CODE REVIEW

**Comment by alissapajer** [comment]
@sellout thanks!

File: connector/src/main/scala/quasar/qscript/MapFunc.scala [hunk 1]
```diff
@@ -757,27 +722,20 @@ object MapFuncs {
...
-  @Lenses final case class ExtractDecade[T[_[_]], A](a1: A) extends Unary[T, A]
   @Lenses final case class ExtractDayOfWeek[T[_[_]], A](a1: A) extends Unary[T, A]
   @Lenses final case class ExtractDayOfYear[T[_[_]], A](a1: A) extends Unary[T, A]
   @Lenses final case class ExtractEpoch[T[_[_]], A](a1: A) extends Unary[T, A]
   @Lenses final case class ExtractHour[T[_[_]], A](a1: A) extends Unary[T, A]
-  @Lenses final case class ExtractIsoDayOfWeek[T[_[_]], A](a1: A) extends Unary[T, A]
-  @Lenses final case class ExtractIsoYear[T[_[_]], A](a1: A) extends Unary[T, A]
-  @Lenses final case class ExtractMicroseconds[T[_[_]], A](a1: A) extends Unary[T, A]
-  @Lenses final case class ExtractMillennium[T[_[_]], A](a1: A) extends Unary[T, A]
-  @Lenses final case class ExtractMilliseconds[T[_[_]], A](a1: A) extends Unary[T, A]
```
Comment (alissapajer) [inline connector/src/main/scala/quasar/qscript/MapFunc.scala]: @sellout was deleting all these map funcs intentional? Or even if it wasn't meant for this diff, was there some thinking that we don't need them anymore?

Comment (sellout) [inline connector/src/main/scala/quasar/qscript/MapFunc.scala]: Ah, that got lumped into this commit accidentally. The intent is that connectors don’t need to implement all of these operations – Eg, `ExtractCentury` can be implemented as `Floor(ExtractYear / 100)`. There’s now `MapFuncDerived` which could have these cases in it, but my thinking at the time was that they would just be handled directly by the SQL² compiler. These changes cover some of the things mentioned in #2169.

**Comment by alissapajer** [comment]
Closing this since I'll be rebasing the commit to exclude the `MapFunc` deletions.

**Pull request closed without merging**